AddonTerraformTemplate/main.tf

```diff
@@ -90,6 +90,18 @@ resource "azurerm_monitor_data_collection_rule_association" "dcra" {
   ]
 }
 
+resource "azurerm_monitor_data_collection_rule_association" "dcraprivatelink" {
+  count = var.useAzureMonitorPrivateLinkScope && azurerm_kubernetes_cluster.k8s.location == azurerm_monitor_workspace.amw.location ? 1 : 0
+  data_collection_endpoint_id = azurerm_monitor_data_collection_endpoint.dce.id
+  target_resource_id  = azurerm_kubernetes_cluster.k8s.id
+  description             = "Association of data collection rule. Deleting this association will break the data collection for this AKS Cluster."
```

Comment:
Name defaults to configurationAccessEndpoint -> [doc](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/monitor_data_collection_rule_association).